### PR TITLE
CI: Add new workflow for Intel SDE

### PR DESCRIPTION
**Pull request by seiko2plus**
To test universal intrinsics against the latest AVX512 extensions.

### CODE REVIEW

**Comment by rgommers** [comment]
This is a very nice CI job - easy to follow along what is going on, and it even runs pretty quickly with emulated instructions. Nice idea, thanks @seiko2plus!

File: .github/workflows/build_test.yml [hunk 1]
```diff
@@ -277,3 +277,32 @@ jobs:
         docker run --rm --interactive -v $(pwd):/numpy the_build /bin/bash -c "
           cd /numpy && python3 runtests.py -n -v -- -k test_simd
         "
+
+  sde_simd_avx512_test:
+    needs: [smoke_test]
```
Comment (mattip) [inline .github/workflows/build_test.yml:282]: Can we add a comment for people who don't know what sde is, something like
```suggestion
    # Intel provides a Software Development Emulator (SDE) that allows emulating
    # various instruction sets. Use it to check different SIMD commands.
    needs: [smoke_test]
```

Comment (seiko2plus) [reply .github/workflows/build_test.yml:285]: done

**Comment by seiko2plus** [comment]
@rgommers, Not that fast, that's why we only use it for testing universal intrinsics. similar to #20206.

**Comment by mattip** [comment]
Thanks @seiko2plus

**Pull request merged**